fortitude/src/rules/mod.rs

```diff
@@ -74,92 +74,92 @@ pub fn code_to_rule(category: Category, code: &str) -> Option<(RuleGroup, Rule)>
-        (Modules, "022") => (RuleGroup::Preview, Ast, modules::accessibility_statements::DefaultPublicAccessibility),
-        (Modules, "031") => (RuleGroup::Preview, Ast, modules::include_statement::IncludeStatement),
-        (Modules, "041") => (RuleGroup::Preview, Ast, modules::file_contents::MultipleModules),
-        (Modules, "042") => (RuleGroup::Preview, Ast, modules::file_contents::ProgramWithModule),
-
-        (Io, "001") => (RuleGroup::Preview, Ast, io::missing_specifier::MissingActionSpecifier),
-        (Io, "011") => (RuleGroup::Preview, Ast, io::magic_io_unit::MagicIoUnit),
-        (Io, "012") => (RuleGroup::Preview, Ast, io::magic_io_unit::NonPortableIoUnit),
-
-        (Readability, "001") => (RuleGroup::Preview, Ast, readability::magic_numbers::MagicNumberInArraySize),
-
-        (Bugprone, "001") => (RuleGroup::Preview, Ast, bugprone::select_default::MissingDefaultCase),
-        (Bugprone, "011") => (RuleGroup::Preview, Ast, bugprone::trailing_backslash::TrailingBackslash),
+        (Error, "000") => (RuleGroup::Stable, None, Default, error::ioerror::IoError),
+        (Error, "001") => (RuleGroup::Stable, Ast, Default, error::syntax_error::SyntaxError),
+        (Error, "011") => (RuleGroup::Stable, None, Default, error::allow_comments::InvalidRuleCodeOrName),
+
+        (Filesystem, "001") => (RuleGroup::Stable, Path, Default,filesystem::extensions::NonStandardFileExtension),
+
+        (Style, "001") => (RuleGroup::Stable, Text, Default, style::line_length::LineTooLong),
+        (Style, "021") => (RuleGroup::Stable, Ast, Default, style::exit_labels::MissingExitOrCycleLabel),
+        (Style, "041") => (RuleGroup::Stable, Ast, Default, style::old_style_array_literal::OldStyleArrayLiteral),
+        (Style, "051") => (RuleGroup::Stable, Ast, Default, style::relational_operators::DeprecatedRelationalOperator),
+        (Style, "061") => (RuleGroup::Stable, Ast, Default, style::end_statements::UnnamedEndStatement),
+        (Style, "071") => (RuleGroup::Stable, Ast, Default, style::double_colon_in_decl::MissingDoubleColon),
+        (Style, "081") => (RuleGroup::Preview, Ast, Default, style::semicolons::SuperfluousSemicolon),
+        (Style, "082") => (RuleGroup::Preview, Ast, Default, style::semicolons::MultipleStatementsPerLine),
+        (Style, "101") => (RuleGroup::Stable, Text, Default, style::whitespace::TrailingWhitespace),
+        (Style, "102") => (RuleGroup::Stable, Ast, Pedantic, style::whitespace::IncorrectSpaceBeforeComment),
+
+        (Typing, "001") => (RuleGroup::Stable, Ast, Default, typing::implicit_typing::ImplicitTyping),
+        (Typing, "002") => (RuleGroup::Stable, Ast, Default, typing::implicit_typing::InterfaceImplicitTyping),
+        (Typing, "003") => (RuleGroup::Stable, Ast, Default, typing::implicit_typing::SuperfluousImplicitNone),
+        (Typing, "004") => (RuleGroup::Preview, Ast, Default, typing::implicit_typing::ImplicitExternalProcedures),
```

Comment:
I'm not sure about this one. I think it's a good idea to do this, but I think it's rare to see people do it in practice. Maybe it's okay as long as it's kept in preview?

Reply:
I think it's rare to see because people don't know about it -- which I think is a good argument for us having it on by default! It's reasonably widely supported, basically just nvfortran and amd-flang/aocc that don't. Sensible to keep it in preview for now though